Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,60 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
+    // Inherited for now.
+    /*
+    int kind { get; set; }
+    string filename { get; set; }
+    int pos { get; set; }
+    int col { get; set; }
+    int line { get; set; }
+    string val { get; set; }
+    bool IsValid { get; }*/
+    string leadingTrivia { get; set; }
+    string trailingTrivia { get; set; }
+  }
+
+  public record Token : IToken {
+    public Token peekedTokens; // Used only internally when the scanner "peeks" tokens. Normallly null at the end of parsing
+    public static readonly IToken NoToken = (IToken)new Token();
+
+    public Token() => this.val = "anything so that it is nonnull";
+
+    public Token(int linenum, int colnum) {
```

Comment:
Shouldn't this also take a value for `val` ?

Reply:
For now, I just copied the token's constructors from Boogie. We could add more constructors based on the need later, but I did not want to overweight this PR.

Reply:
> I did not want to overweight this PR

One option could have been for the PR introduce Dafny.Token but not do anything with trivia, and make `Dafny.Token` a class that follows our coding standards. I wouldn't expect `Dafny.Token` to be mutable for example, and while the setters are not often called it would take some extra changes to remove them.